### PR TITLE
Add ability to use "Grouped" CV instances from scikit-learn

### LINKED ISSUE #181: Allow for group-based cross-validation objects from scikit-learn
**Problem Description**
Probatus feature elimination (e.g. `ShapRFECV`) currently does not allow for cross-validation objects which take `groups` variables (e.g. [StratifiedGroupKFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedGroupKFold.html))

**Desired Outcome**
It would be great if this feature could be implemented as those `groups` can be used to prevent data leakage in (e.g.) the case where multiple samples from the same customer are available and therefore should be either only in the training or the test set but not in both. 

**Solution Outline**
The fix to this _should_ be quite simple and can follow the implementation of `scikit-learn`'s [RFECV](https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/feature_selection/_rfe.py#L635): One would need to add a `groups` variable (default: `None`) to the `fit`/`fit_compute` methods of `ShapRFECV` and pass it through to [self.cv.split](https://github.com/ing-bank/probatus/blob/56afdc916d816fdc31fd4e71f8eb29413ff5a829/probatus/feature_elimination/feature_elimination.py#L549)

### PR DESCRIPTION
**Pull request by PaulZhutovsky**
closes #181 

Allows the usage of Group CV objects in `ShapRFECV`.

### CODE REVIEW

**Review by ClaudioSalvatoreArcidiacono — APPROVED** [review]
Looks good to me.

File: probatus/feature_elimination/feature_elimination.py [hunk 1]
```diff
@@ -402,6 +402,7 @@ def fit(
         self,
         X,
         y,
+        groups=None,
         sample_weight=None,
         columns_to_keep=None,
         column_names=None,
```
Comment (ClaudioSalvatoreArcidiacono) [inline probatus/feature_elimination/feature_elimination.py:408]: I would add the new param as a last parameter to improve backward compatibility
```suggestion
        sample_weight=None,
        columns_to_keep=None,
        column_names=None,
        groups=None,
```

**Review by ClaudioSalvatoreArcidiacono — CHANGES_REQUESTED** [review]

File: probatus/feature_elimination/feature_elimination.py (hunk 1 above)
Comment (PaulZhutovsky) [reply probatus/feature_elimination/feature_elimination.py]: I made the change, thanks! I was following the argument order for [RFECV](https://github.com/scikit-learn/scikit-learn/blob/37ac6788c9504ee409b75e5e24ff7d86c90c2ffb/sklearn/feature_selection/_rfe.py#L635) but you are right it makes more sense to be backward compatible here.

**Review by ClaudioSalvatoreArcidiacono — APPROVED** [review]

**Comment by PaulZhutovsky** [comment]
The tests fail but the reason seems to be that pip is not able to resolve the dependencies under Python 3.7 which was mentioned by @operte.

**Comment by timvink** [comment]
The related issue is #185

**Comment by operte** [comment]
The failed checks are not related to this PR, so I will merge it.
